File: lab2.py
def map():

    return

n = 0
alfabeto = []
while(True):
    print("")
    print("[1] Asociar alfabeto con Zn.")
    print("[2] Vigenènere encipher.")
    print("[3] Vigenènere decipher.")
    print("[0] Salir")
    opc = int(input("Selecciona una opción: "))
    print("")

    match(opc):
        case 1:
            n = int(input("Ingresa el tamaño del alfabeto"))
            for i in range(n):
                c = input(f"Ingresa el elemento {i} del alfabeto: ")
                alfabeto.append(c)
            
            
        case 2:
            break

        case 3:
            break

        case 0:
            break
            
        case _:
            print("¡Ingresa una opción válida! :(")
            print("_____________________________")
        
    print("")

File: lab2.cpp
#include <iostream>
#include <map>
#include <vector>
#include <string>
using namespace std;

int main() {

    int n;
    map<char, int> alfabeto;

    while(true) {
        int opc;
        char c;
        string s, m, K;
        cout << "[1] Ingresar alfabeto." << endl;
        cout << "[2] Vigenènere encipher." << endl;
        cout << "[3] Vigenènere decipher." << endl;
        cout << "[0] Salir" << endl;
        cout << "Selecciona una opción: ";
        cin >> opc;
        cout << endl;

        switch(opc) {
            case 1:
                cout << "Ingresa el tamaño del alfabeto: ";
                cin >> n;
                
                for(int i=0; i<n; i++) {
                    cout << "Ingresa el elemento " << i+1 << ": ";
                    cin.ignore();
                    c = getchar();
                    alfabeto[c] = i;
                }

                for(auto it : alfabeto) {
                    cout << it.first << " - " << it.second << endl;
                }
                break;

            case 2:
                cout << "Ingresa la llave K: ";
                cin.ignore();
                getline(cin, K);

                if(K.length() < 5) {
                    cout << "ERROR: |K| < 5" << endl;
                    exit(1);
                }
                for(auto it : K) {
                    if(!alfabeto.count(it)) {
                        cout << "ERROR: " << it << " ∉ Σ*" << endl;
                        exit(1);
                    }
                }

                cout << "Ingresa el mensaje m: ";
                getline(cin, m);
                if(m.length() < 30) {
                    cout << "ERROR: |m| < 30" << endl;
                }
                for(auto it : m) {
                    if(!alfabeto.count(it)) {
                        cout << "ERROR: " << it << " ∉ Σ*" << endl;
                        exit(1);
                    }
                }
                
                break;

            case 3:
                break;

            case 0:
                break;
                
            default:
                cout << ("¡Ingresa una opción válida! :(") << endl;
                cout << ("_____________________________") << endl;
        }
        cout << endl;
    }
    return 0;
}